ambry-store/src/main/java/com/github/ambry/store/LogSegment.java

```diff
@@ -154,6 +155,9 @@ class LogSegment implements Read, Write {
     } catch (FileNotFoundException e) {
       throw new StoreException("File not found while creating log segment [" + file.getAbsolutePath() + "]", e,
           StoreErrorCodes.File_Not_Found);
+    } catch (EOFException e) {
```

Comment:
Good to me.
Currently even we don't catch it, it goes with the generic Exception. With generic exception, we also generate the same StoreException with Log_File_Format_Error as well, right?

Reply:
Not really, EOFException is a subclass of IOException, it will be caught by the catch statement and then return Uknown_Error code.